File: Iterator/Menu.cs
using System;
using System.Collections;
using System.Linq;
using System.Text;

namespace Iterator
{

    public class Menu : IMenu
    {
        string name;
        public string Name
        {
            get { return name; }
            set { name = value; }
        }

        string description;
        public string Description
        {
            get { return description; }
            set { description = value; }
        }

        ArrayList children = new ArrayList();

        public Menu(String name, String description)
        {
            this.Name = name;
            this.Description = description;
        }

        public void Print()
        {
            Console.WriteLine("Menu\n{0}, {1}\n---------------------",
                this.Name, this.Description);

            foreach (IMenu child in this.children)
                child.Print();
        }

        public void Add(IMenu menu)
        {
            this.children.Add(menu);
        }
    }
}

File: Composite/src/com/jkhines/DesignPatterns/CompositePattern/Menu.java
package com.jkhines.DesignPatterns.CompositePattern;

import java.util.*;

public class Menu extends MenuNode {
	private String name;
	private String description;
	private ArrayList<MenuNode> children;
	
	public Menu(String name, String description) {
		this.name = name;
		this.description = description;
		children = new ArrayList<MenuNode>();
	}
	
	@Override
	public void add(MenuNode child) {
		children.add(child);
	}
	
	@Override
	public void remove(MenuNode child) {
		children.remove(child);
	}
	
	@Override
	public MenuNode getChild(int i) {
		return children.get(i);
	}
	
	@Override
	public String getName() {
		return name;
	}
	
	@Override
	public String getDescription() {
		return description;
	}
	
	@Override
	public void print() {
		System.out.println(name);
		System.out.println(description);
		
		for (MenuNode node : children)
			node.print();
	}

	@Override
	public Iterator<MenuNode> iterator() {
		return children.iterator();
	}
}
